### ISSUE TITLE
arkclient fails to start on ubuntu 17.04

### ISSUE THREAD

**Issue by Cerv4ntez** [ArkEcosystem/ark-desktop#254]
Error message:
A JavaScript error occurred in the main process
Uncaught Exception:
Error: /tmp/.org.chromium.Chromium.036OUi: failed to map segment from shared object
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:187:18)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/opt/ArkClient/resources/app.asar/node_modules/node-hid/nodehid.js:9:15)
    at Object.<anonymous> (/opt/ArkClient/resources/app.asar/node_modules/node-hid/nodehid.js:109:3)

**Comment by fix** [comment]
node-hid build does not run on your os for some reason.
Can you give more info on your os? 32 or 64?

**Comment by Cerv4ntez** [comment]
sure, whatever you need, 64bit running on kernel 4.13, do you need the version of any specific package?

**Comment by Cerv4ntez** [comment]
so, i solved this by myself. in case someone else runs into this(he or she probably followed a guide to harden security on linux systems):
this is a permissions problem, when the setuid bit is unset in tmp, you run into this, fix it by either setting it(in fstab) or defining the TMPDIR env variable

**Comment by fix** [comment]
Great you managed to fixed it.
mm maybe there is something to be done then by specifying a tmp folder at root of home directory. something to see with electron default settings.

**Comment by YellowSharkMT** [comment]
Thanks @Cerv4ntez! You helped me identify a problem with installing `node-hid` on my system (Ubuntu 16.04). I was unable to launch `arkclient`, was experiencing exact same error as you, and I discovered that I was unable to install/re-install `node-gyp` on my system - I was getting stuck in a loop with this error message:

    gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/node-hid/.node-gyp"
    gyp WARN EACCES user "root" does not have permission to access the dev dir "/usr/lib/node_modules/node-hid/.node-gyp/8.11.1"

Your mention of the permissions problem on the temp directory triggered me to consider the fact that I've got my system configured with the `/tmp` dir located on a separate partition, mounted with the `nosuid` option.

I didn't feel like changing that, so I resorted to making a new, temporary temp dir, exporting TMPDIR, and re-installing node-hid - it all went perfect:

    $ sudo mkdir ~/super_duper_temp_dir
    $ sudo chmod -R 777 ~/super_duper_temp_dir
    $ export TMPDIR=~/super_duper_tempdir
    $ sudo npm install -g node-hid # previously failed, now it works great....

After that, I was able to launch `arkclient` with no problems. Just wanted to say thanks very much!

**Comment by Cerv4ntez** [comment]
Hi, I'm glad this helped you, also, very appreciative of your appreciation 😉Am 05.04.2018 18:52 schrieb The Yellow Shark <notifications@github.com>:Thanks @Cerv4ntez! You helped me identify a problem with installing node-hid on my system (Ubuntu 16.04). I was unable to launch arkclient, was experiencing exact same error as you, and I discovered that I was unable to install/re-install node-gyp on my system - I was getting stuck in a loop with this error message:
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/node-hid/.node-gyp"
gyp WARN EACCES user "root" does not have permission to access the dev dir "/usr/lib/node_modules/node-hid/.node-gyp/8.11.1"

Your mention of the permissions problem on the temp directory triggered me to consider the fact that I've got my system configured with the /tmp dir located on a separate partition, mounted with the nosuid option.
I didn't feel like changing that, so I resorted to making a new, temporary temp dir, exporting TMPDIR, and re-installing node-hid - it all went perfect:
$ sudo mkdir ~/super_duper_temp_dir
$ sudo chmod -R 777 ~/super_duper_temp_dir
$ export TMPDIR=~/super_duper_tempdir
$ sudo npm install -g node-hid # previously failed, now it works great....

After that, I was able to launch arkclient with no problems. Just wanted to say thanks very much!
